.travis.yml

```diff
@@ -46,11 +47,38 @@ jobs:
             repo: open-policy-agent/gatekeeper
             branch: master
             tags: false
-        - provider: script
-          skip_cleanup: true
-          script: make travis-dev-release
-          on:
-            repo: open-policy-agent/gatekeeper
-            branch: master
-            tags: true
-
+    - stage: "build, release"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request AND tag IS present
+      script:
+      - make manager
+      # Push new release version to registry
+      - make travis-dev-release IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG} VERSION=${TRAVIS_TAG}
+    - stage: "verify release"
+      name: "e2e on released image"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request AND tag IS present
+      before_script:
+      - KIND_VERSION=0.4.0
+      - KUSTOMIZE_VERSION=3.0.2
+      # Download and install kind
+      - curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output kind && chmod +x kind && sudo mv kind /usr/local/bin/
+      # Create kind cluster
+      - |
+        if [ $(kind get clusters) ]; then
+          export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+        else
+          kind create cluster
+        fi
+      # Get kubeconfig
+      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+      # Update manager_image_patch.yaml with latest release tag (git tag)
+      - make docker-build IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG}
+      # Update deployment yamls with the released tag
+      - make release-deploy NEWVERSION=${TRAVIS_TAG}
+      # Download and install kubectl
+      - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && chmod +x ./kubectl && sudo mv kubectl /usr/local/bin/
+      # Download and install kustomize
+      - curl -L https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64 --output kustomize && chmod +x kustomize && sudo mv kustomize /usr/local/bin/
+      script:
+      - make deploy
```

Comment:
Where are we adding the image to `kind`?

Reply:
We dont need to add the image to `kind` as the yaml we are deploying actually uses the new released image on quay. we only need to add image to `kind` if it's a local image.

Reply:
Ah, nice!